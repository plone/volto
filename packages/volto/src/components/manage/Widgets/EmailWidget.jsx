/**
 * EmailWidget component.
 * @module components/manage/Widgets/EmailWidget
 */

import { FormFieldWrapper } from '@plone/volto/components';
import PropTypes from 'prop-types';
import React from 'react';
import { Input } from 'semantic-ui-react';

/** EmailWidget, a widget for email addresses
 *
 * To use it, in schema properties, declare a field like:
 *
 * ```jsx
 * {
 *  title: "Email",
 *  widget: 'email',
 * }
 * ```
 */
const EmailWidget = (props) => {
  const {
    id,
    value,
    onChange,
    onBlur,
    onClick,
    minLength,
    maxLength,
    placeholder,
    isDisabled,
  } = props;
  const inputId = `field-${id}`;

  return (
    <FormFieldWrapper {...props} className="email">
      <Input
        id={inputId}
        name={id}
        type="email"
        value={value || ''}
        disabled={isDisabled}
        placeholder={placeholder}
        onChange={({ target }) =>
          onChange(id, target.value === '' ? undefined : target.value)
        }
        onBlur={({ target }) =>
          onBlur(id, target.value === '' ? undefined : target.value)
        }
        onClick={() => onClick()}
        minLength={minLength || null}
        maxLength={maxLength || null}
      />
    </FormFieldWrapper>
  );
};

/**
 * Property types
 * @property {Object} propTypes Property types.
 * @static
 */
EmailWidget.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.arrayOf(PropTypes.string),
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  onClick: PropTypes.func,
  minLength: PropTypes.number,
  maxLength: PropTypes.number,
  placeholder: PropTypes.string,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
EmailWidget.defaultProps = {
  description: null,
  required: false,
  error: [],
  value: null,
  onChange: () => {},
  onBlur: () => {},
  onClick: () => {},
  minLength: null,
  maxLength: null,
};

export default EmailWidget;
