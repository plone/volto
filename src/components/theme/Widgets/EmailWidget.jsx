import React from 'react';
import cx from 'classnames';
import { UniversalLink } from '@plone/volto/components';

const EmailWidget = ({ value, children, className }) =>
  value ? (
    <UniversalLink
      href={'mailto:' + value}
      className={cx(className, 'email', 'widget')}
    >
      {children ? children(value) : value}
    </UniversalLink>
  ) : (
    ''
  );

export default EmailWidget;
