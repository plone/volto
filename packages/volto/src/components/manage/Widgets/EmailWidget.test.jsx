import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import EmailWidget from './EmailWidget';

const mockStore = configureStore();

test('renders an email widget component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <EmailWidget
        id="test-email"
        title="My Email"
        fieldSet="default"
        onChange={() => {}}
        onBlur={() => {}}
        onClick={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
