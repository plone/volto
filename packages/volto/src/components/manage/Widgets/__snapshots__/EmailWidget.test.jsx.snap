// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an email widget component 1`] = `
<div
  className="inline field email field-wrapper-test-email"
>
  <div
    className="ui grid"
  >
    <div
      className="stretched row"
    >
      <div
        className="four wide column"
      >
        <div
          className="wrapper"
        >
          <label
            htmlFor="field-test-email"
            id="fieldset-default-field-label-test-email"
          >
            My Email
          </label>
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="ui input"
        >
          <input
            id="field-test-email"
            maxLength={null}
            minLength={null}
            name="test-email"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
